distributed/deploy/adaptive.py

```diff
@@ -81,16 +100,21 @@ class Adaptive(AdaptiveCore):
     specified in the dask config under the distributed.adaptive key.
     '''
 
+    interval: float | None
+    periodic_callback: PeriodicCallback | None
+    #: Whether this adaptive strategy is periodically adapting
+    state: AdaptiveStateState
+
     def __init__(
         self,
-        cluster=None,
-        interval=None,
-        minimum=None,
-        maximum=None,
-        wait_count=None,
-        target_duration=None,
-        worker_key=None,
-        **kwargs,
+        cluster: Cluster,
+        interval: str | float | timedelta | None = None,
+        minimum: int | None = None,
+        maximum: float | None = None,
```

Comment:
Hm, this is a little confusing given that we actually either want `int` or `math.inf`, per https://github.com/dask/distributed/blob/8bafad5450592d9a647080589af1ca3072cc5341/distributed/deploy/adaptive_core.py#L112-L113

Reply:
I don't have strong feelings here, this just follow the numeric tower in [PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower), i.e., `int | float` can be simplified to `float`. Unfortunately there isn't a canonical `INF` literal that I'm aware of, instead we have `math.inf`, `float("inf")`, `np.inf`, and more. Happy to change it back if people find it confusing.

Reply:
Intuitively I would say that `None` is equivalent to `math.inf` in this case. I expect most users would assume that setting the value to `None` would not set an upper bound. They might even assume that setting it to `0` does the same.

Should we attempt to define an `INF` literal in `dask.typing` that we can use here?

Reply:
Sounds reasonable but let's do that in a different PR. I'll roll my changes to the type back for now.

Reply:
Note that `None` is not equivalent to `math.inf` in this case. Instead, we read the config value if `None` is provided.